### ISSUE TITLE
Panic in intervaltree

### ISSUE THREAD

**Issue by sfackler** [gimli-rs/addr2line#69]
This isn't an issue in addr2line itself, but intervaltree doesn't say what repo it lives in so I have no idea how to file an issue against it 😢 

```
thread 'main' panicked at 'assertion failed: !nodes.is_empty()', /Users/sfackler/.cargo/registry/src/github.com-1ecc6299db9ec823/intervaltree-0.2.0/src/lib.rs:100:8
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
   1: std::panicking::default_hook::{{closure}}
   2: std::panicking::default_hook
   3: std::panicking::rust_panic_with_hook
   4: std::panicking::begin_panic
   5: <intervaltree::IntervalTree<K, V>>::update_max
   6: <intervaltree::IntervalTree<K, V> as core::iter::traits::FromIterator<I>>::from_iter
   7: core::iter::iterator::Iterator::collect
   8: <addr2line::Context<R>>::parse_functions
   9: addr2line::main
  10: __rust_maybe_catch_panic
  11: std::rt::lang_start
  12: main
```

It specifically looks like its collect implementation is broken for empty iterators.

**Comment by sfackler** [comment]
@main--

**Comment by main--** [comment]
>intervaltree doesn't say what repo it lives in so I have no idea how to file an issue against it 😢 

Sorry, I'm usually too lazy to fill that in, as you may have guessed it's https://github.com/main--/rust-intervaltree.

This is indeed an edge case I haven't considered, fixing this right now.

**Comment by main--** [comment]
Fixed in https://github.com/main--/rust-intervaltree/commit/4757300ca1fb4f0c2ddf64b7d060dc0b0e3b0757, published in version 0.2.1.

**Comment by sfackler** [comment]
Thanks!
